src/unity/lib/annotation/annotation_base.hpp

```diff
@@ -74,6 +105,16 @@ class AnnotationBase {
   void _addIndexColumn();
   void _checkDataSet();
   void _reshapeIndicies(size_t &start, size_t &end);
+
+private:
+  /* A little trick to overload the `__serialize_proto` function at compile time
+   * so I don't have to define that for every Annotation::Specification type.
+   *
+   * Using the SFINAE method: https://en.cppreference.com/w/cpp/language/sfinae
+   */
+  template <typename T, typename = typename std::enable_if<std::is_base_of<
+                            ::google::protobuf::MessageLite, T>::value>::type>
+  std::string __serialize_proto(T message);
```

Comment:
Nice! I see the definition is elsewhere though. Won't that potentially result in undefined symbols at link time? For templates, I've always seen the definition in the header file where it's declared.

Reply:
Ya, so I ran into this issue when compiling this template. However I believe there was an explanation here:

https://stackoverflow.com/questions/495021/why-can-templates-only-be-implemented-in-the-header-file
https://isocpp.org/wiki/faq/templates#templates-defn-vs-decl